frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BasePullBranchFastForwardOnlyAction.java

```diff
@@ -86,7 +88,11 @@ public void actionPerformed(AnActionEvent anActionEvent) {
 
       val currentBranchName = Option.of(gitRepository.getCurrentBranch()).map(b -> b.getName()).getOrNull();
       if (localBranchName.equals(currentBranchName)) {
-        doPullCurrentBranchFastForwardOnly(project, gitRepository, remoteBranch);
+        if (FetchAllRemotesAction.isUpToDate()) {
+          doPullCurrentBranchFastForwardOnly(project, gitRepository, remoteBranch);
```

Comment:
Hmm the factoring here is a bit confusing... since we're checking `isUpToDate` here on a different level in this case than in the other case 🤔